Problem with Linux64 Build

```
Build FAILED.

Warnings:

/home/szope/Projects/Vocaluxe/Vocaluxe.sln:  warning : Ignoring vcproj 'PitchTracker'.
/home/szope/Projects/Vocaluxe/Vocaluxe.sln:  warning : Failed to find project 54a48af7-515c-4edd-b395-2bffa096050a

/home/szope/Projects/Vocaluxe/Vocaluxe.sln (default targets) ->
(Build target) ->
/home/szope/Projects/Vocaluxe/VocaluxeLib/VocaluxeLib.csproj (default targets) ->
/usr/lib/mono/xbuild/14.0/bin/Microsoft.CSharp.targets (CoreCompile target) ->

	Menu/SongMenu/CSongMenuList.cs(639,17): warning CS0219: The variable `totalSongNumber' is assigned but its value is never used
	Menu/SongMenu/CSongMenuFramework.cs(292,21): warning CS0219: The variable `check' is assigned but its value is never used

Errors:

/home/szope/Projects/Vocaluxe/Vocaluxe.sln (default targets) ->
(Build target) ->
/home/szope/Projects/Vocaluxe/Vocaluxe/Vocaluxe.csproj (default targets) ->
(EnsureNuGetPackageBuildImports target) ->

	/home/szope/Projects/Vocaluxe/Vocaluxe/Vocaluxe.csproj: error : This project references NuGet package(s) that are missing on this computer. Rebuild for automatic package restore. The missing file is ../packages/vocaluxe.dependencies.aforgevideo.1.0.1/build/vocaluxe.dependencies.aforgevideo.targets.

/home/szope/Projects/Vocaluxe/Vocaluxe.sln (default targets) ->
(Build target) ->
/home/szope/Projects/Vocaluxe/Vocaluxe/Vocaluxe.csproj (default targets) ->
/usr/lib/mono/xbuild/14.0/bin/Microsoft.Common.targets (_RecordCleanFile target) ->

	/usr/lib/mono/xbuild/14.0/bin/Microsoft.Common.targets: error : Could not find a part of the path "/home/szope/Projects/Vocaluxe/Vocaluxe/obj/x64/ReleaseLinux/Vocaluxe.csproj.FilesWrittenAbsolute.txt".  at System.IO.FileStream..ctor (System.String path, System.IO.FileMode mode, System.IO.FileAccess access, System.IO.FileShare share, System.Int32 bufferSize, System.Boolean anonymous, System.IO.FileOptions options) [0x001cf] in <8f2c484307284b51944a1a13a14c0266>:0 
  at System.IO.FileStream..ctor (System.String path, System.IO.FileMode mode, System.IO.FileAccess access, System.IO.FileShare share, System.Int32 bufferSize, System.IO.FileOptions options, System.String msgPath, System.Boolean bFromProxy, System.Boolean useLongPath, System.Boolean checkHost) [0x00000] in <8f2c484307284b51944a1a13a14c0266>:0 
  at (wrapper remoting-invoke-with-check) System.IO.FileStream:.ctor (string,System.IO.FileMode,System.IO.FileAccess,System.IO.FileShare,int,System.IO.FileOptions,string,bool,bool,bool)
  at System.IO.StreamWriter.CreateFile (System.String path, System.Boolean append, System.Boolean checkHost) [0x00022] in <8f2c484307284b51944a1a13a14c0266>:0 
  at System.IO.StreamWriter..ctor (System.String path, System.Boolean append, System.Text.Encoding encoding, System.Int32 bufferSize, System.Boolean checkHost) [0x00061] in <8f2c484307284b51944a1a13a14c0266>:0 
  at System.IO.StreamWriter..ctor (System.String path, System.Boolean append, System.Text.Encoding encoding, System.Int32 bufferSize) [0x00000] in <8f2c484307284b51944a1a13a14c0266>:0 
  at System.IO.StreamWriter..ctor (System.String path, System.Boolean append) [0x00008] in <8f2c484307284b51944a1a13a14c0266>:0 
  at (wrapper remoting-invoke-with-check) System.IO.StreamWriter:.ctor (string,bool)
  at Microsoft.Build.Tasks.WriteLinesToFile.Execute () [0x00034] in <063b27d92e604f699a7eab9b5479d82b>:0 

	 4 Warning(s)
	 2 Error(s)
```

Im getting this kind of error after make

Linux S161 3.16.0-4-amd64 #1 SMP Debian 3.16.36-1+deb8u2 (2016-10-19) x86_64 GNU/Linux

I now tried to make the latest nightly of Vocaluxe 0.4.0.beta1-31-g2e6b388 on Linux Mint 18.1 (which is based on Ubuntu 16.04, if I remember it correctly). Well, it stopped me here:
```sh
make -C PitchTracker
make[1]: Verzeichnis „/home/stefan/Downloads/Vocaluxe-Nightly/PitchTracker“ wird betreten
gcc -shared -o libPitchTracker.dll.so -fPIC compatibility.o dywapitchtrack/dywapitchtrack.o dywapitchtrack/ptDyWa.o FFT/FFT.o FFT/RealFFTf.o Helper.o performous/pitch.o ptAKF.o PitchWrapper.o
strip libPitchTracker.dll.so
cp libPitchTracker.dll.so ../Output/
make[1]: Verzeichnis „/home/stefan/Downloads/Vocaluxe-Nightly/PitchTracker“ wird verlassen
make -C Vocaluxe/Lib/Video/Acinerella
make[1]: Verzeichnis „/home/stefan/Downloads/Vocaluxe-Nightly/Vocaluxe/Lib/Video/Acinerella“ wird betreten
gcc -c -fPIC acinerella.c -I /usr/local/include
acinerella.c: In function ‘cpymetadata’:
acinerella.c:344:9: warning: implicit declaration of function ‘strncpy_s’ [-Wimplicit-function-declaration]
         strncpy_s(tar, len, entry->value, len);
         ^
acinerella.c: In function ‘ac_get_stream_info’:
acinerella.c:517:42: error: ‘AVStream {aka struct AVStream}’ has no member named ‘codecpar’
     switch (self->pFormatCtx->streams[nb]->codecpar->codec_type) {
                                          ^
acinerella.c:524:46: error: ‘AVStream {aka struct AVStream}’ has no member named ‘codecpar’
                 self->pFormatCtx->streams[nb]->codecpar->width;
                                              ^
acinerella.c:526:46: error: ‘AVStream {aka struct AVStream}’ has no member named ‘codecpar’
                 self->pFormatCtx->streams[nb]->codecpar->height;
                                              ^
acinerella.c:529:46: error: ‘AVStream {aka struct AVStream}’ has no member named ‘codecpar’
                 self->pFormatCtx->streams[nb]->codecpar->sample_aspect_ratio.num;
                                              ^
acinerella.c:531:46: error: ‘AVStream {aka struct AVStream}’ has no member named ‘codecpar’
                 self->pFormatCtx->streams[nb]->codecpar->sample_aspect_ratio.den;
                                              ^
acinerella.c:551:46: error: ‘AVStream {aka struct AVStream}’ has no member named ‘codecpar’
                 self->pFormatCtx->streams[nb]->codecpar->sample_rate;
                                              ^
acinerella.c:553:46: error: ‘AVStream {aka struct AVStream}’ has no member named ‘codecpar’
                 self->pFormatCtx->streams[nb]->codecpar->channels;
                                              ^
acinerella.c:556:50: error: ‘AVStream {aka struct AVStream}’ has no member named ‘codecpar’
             switch (self->pFormatCtx->streams[nb]->codecpar->format) {
                                                  ^
acinerella.c: In function ‘ac_create_video_decoder’:
acinerella.c:683:5: error: unknown type name ‘AVCodecParameters’
     AVCodecParameters* pCodecParam = ((lp_ac_data)(pacInstance))->pFormatCtx->streams[nb]->codecpar;
     ^
acinerella.c:683:90: error: ‘AVStream {aka struct AVStream}’ has no member named ‘codecpar’
     AVCodecParameters* pCodecParam = ((lp_ac_data)(pacInstance))->pFormatCtx->streams[nb]->codecpar;
                                                                                          ^
acinerella.c:688:47: error: request for member ‘codec_id’ in something not a structure or union
               avcodec_find_decoder(pCodecParam->codec_id))) {
                                               ^
acinerella.c:693:5: warning: implicit declaration of function ‘avcodec_parameters_to_context’ [-Wimplicit-function-declaration]
     avcodec_parameters_to_context(pDecoder->pCodecCtx, pCodecParam);
     ^
acinerella.c: In function ‘ac_create_audio_decoder’:
acinerella.c:741:5: error: unknown type name ‘AVCodecParameters’
     AVCodecParameters* pCodecParam = self->pFormatCtx->streams[nb]->codecpar;
     ^
acinerella.c:741:67: error: ‘AVStream {aka struct AVStream}’ has no member named ‘codecpar’
     AVCodecParameters* pCodecParam = self->pFormatCtx->streams[nb]->codecpar;
                                                                   ^
acinerella.c:744:60: error: request for member ‘codec_id’ in something not a structure or union
     ERR(pDecoder->pCodec = avcodec_find_decoder(pCodecParam->codec_id));
                                                            ^
acinerella.c:41:15: note: in definition of macro ‘ERR’
         if (!(E)) {     \
               ^
acinerella.c: In function ‘ac_decode_video_package’:
acinerella.c:816:8: warning: implicit declaration of function ‘avcodec_send_packet’ [-Wimplicit-function-declaration]
     if(avcodec_send_packet(pDecoder->pCodecCtx, pkt->pPack) != 0)
        ^
acinerella.c:821:9: warning: implicit declaration of function ‘avcodec_receive_frame’ [-Wimplicit-function-declaration]
     if (avcodec_receive_frame(pDecoder->pCodecCtx, pDecoder->pFrame) != 0)
         ^
makefile:2: die Regel für Ziel „acinerella“ scheiterte
make[1]: *** [acinerella] Fehler 1
make[1]: Verzeichnis „/home/stefan/Downloads/Vocaluxe-Nightly/Vocaluxe/Lib/Video/Acinerella“ wird verlassen
makefile:4: die Regel für Ziel „all“ scheiterte
make: *** [all] Fehler 2
```

--- a/HowToBuildLinux.md
+++ b/HowToBuildLinux.md
@@ -1,4 +1,46 @@
 # Build Vocaluxe - HowTo (Linux)
 
-* Clone the repository (see [this](https://github.com/Vocaluxe/Vocaluxe/blob/develop/HowToBuildWin.md) for help)
-* Navigate to the cloned repository and execute `make`. Make sure that all dependencies are installed and the process completes without errors.
+Runing Vocaluxe on Linux is not offically supported yet -but you can try it yourself.  
+(the problematic part is fullfilling the runtime dependencies)
+
+
+* Install ffmpeg 3.2 (a quick and dirty compile script can be found here: [installFFmpeg.sh](https://github.com/lukeIam/Vocaluxe/blob/travis/.travis/installFFmpeg.sh) )
+* Get sure your gcc and g++ version is >= 4.8
+* Install at least mono 5.8.0 ([HowTo install mono](http://www.mono-project.com/download/stable/))
+* Clone the repository
+You want to use this branch as it contains some linux specific changes   
+(compiles on Ubuntu 14.04): [travis@lukeIam/Vocaluxe](https://github.com/lukeIam/Vocaluxe/tree/travis)
+* Navigate to the travis branch and execute (commands from [.travis.yml](https://github.com/lukeIam/Vocaluxe/blob/travis/.travis.yml)):
+```bash
+chmod ugo+x ./.build/linuxPostBuildEvent.sh
+chmod ugo+x ./.build/linuxPreBuildEvent.sh
+chmod ugo+x ./.travis/gitDescribe.sh
+wget https://dist.nuget.org/win-x86-commandline/latest/nuget.exe -P ./.travis/
+
+config=Linux platform=x64
+
+# ./.travis/gitDescribe.sh
+mono ./.travis/nuget.exe restore
+make -C PitchTracker
+make -C Vocaluxe/Lib/Video/Acinerella
+msbuild /p:Configuration=Release$config /p:Platform=$platform /p:TargetFrameworkVersion=v4.7 Vocaluxe.sln
+```
+* The build should complete without errors
+* change to the Output directory:
+```sh
+cd Output
+```
+* Start Vocaluxe:
+```bash
+mono Vocaluxe.exe
+```
+* Vocaluxe will start (you will see the splash screen) and then crash because runtime dependencies are missing    
+Required dependencies (by heart - can be incorrect and incomplete):  
+  - gstreamer 1.0
+  - ffmpeg 3.2 (maybe we can use the build script which builds the libs at the moment)
+  - hidapi
+  - libgstreamersharpglue-1.0.0
+  - portaudio
+  - ?
+  
+  * If yo make some progress we would be happy if you share it with us in a pull request/issue.

